### ISSUE TITLE
Fitness Score

### ISSUE THREAD

**Issue by rhalder2023** [ultralytics/ultralytics#15821]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,
So i was under the assumption that fitness score lies between 0 and 1. In my tuning one of the iterations got a score of 1.19, is it possible? or is something wrong?

**Comment by glenn-jocher** [comment]
That's quite fit, good job!

**Comment by Y-T-G** [comment]
Yes, it's possible.

**Comment by rhalder2023** [comment]
> That's quite fit, good job!

Thank You, so it is possible

**Comment by Y-T-G** [comment]
I guess the range would depend on your weights.

For example, the weights are 0.1 and 0.9 for `mAP50` and `mAP50-95` respectively. And the box, mask and keypoints have separate fitness scores that are summed:
https://github.com/ultralytics/ultralytics/blob/848c5cd5062f320dbb2fdcfa1602c6144e23398b/ultralytics/utils/metrics.py#L1004

https://github.com/ultralytics/ultralytics/blob/848c5cd5062f320dbb2fdcfa1602c6144e23398b/ultralytics/utils/metrics.py#L1145

So the maximum would be `0.1 * 1.0 + 0.9 * 1.0 = 1.0` if you use `Detect` model. But `2.0` if you used `Segment` or `Pose`.

**Comment by rhalder2023** [comment]
> I guess the range would depend on your model.
> 
> For example, the weights are 0.1 and 0.9 for `mAP50` and `mAP50-95` respectively. And the box, mask and keypoints have separate fitness scores that are summed:
> 
> https://github.com/ultralytics/ultralytics/blob/848c5cd5062f320dbb2fdcfa1602c6144e23398b/ultralytics/utils/metrics.py#L1004
> 
> https://github.com/ultralytics/ultralytics/blob/848c5cd5062f320dbb2fdcfa1602c6144e23398b/ultralytics/utils/metrics.py#L1145
> 
> So the maximum would be `0.1 * 1.0 + 0.9 * 1.0 = 1.0` if you use `Detect` model. But `2.0` if you used `Segment` or `Pose`.

Ooooohhhhhh, makes sense, since im using the segment model, wait so if max is 2, my model can definitely get better. Sigh.

But thank, i totally understand why segment and pose would possibly get more than 1

**Comment by rhalder2023** [comment]
also

**Comment by rhalder2023** [comment]
Also, I had one tiny question, the weights file generated by the tuner, are the weights that was trained till the epochs mentioned in the tuner right? for example if i said I wanted to tune 10 epochs for 100 iterations, after tuning lets say the best.pt comes from the 64th iteration. So that weights file contains the model weights trained till the 10th epoch with those hyperparameters right?

**Comment by Y-T-G** [comment]
It's the best.pt from all the iterations. Not necessarily from the last epoch since it might have had a better fitness score before the last epoch of an iteration.
